server/routes/config/ConfigRoutes.ts

```diff
@@ -24,36 +41,51 @@ export default class ConfigRoutes {
     )
     if (hasReadOnlyNomisConfigAccess) {
       return res.render('pages/config/index', {
-        model: new ConfigurationViewModel(activePrisons, readOnlyPrisonResults),
+        model: new ConfigurationViewModel(
+          activePrisons,
+          readOnlyPrisonResults,
+          readOnlyChanges,
+          notReadOnlyChanges,
+          updatedScreen,
+        ),
       })
     }
     return res.redirect('/')
   }
 
+  public update: RequestHandler = async (req, res): Promise<void> => {
+    req.flash('id', req.query.id as string)
+    req.flash('readOnly', req.query.readonly as string)
+    req.flash('notReadOnly', req.query.notreadonly as string)
+    return res.redirect(`/config`)
+  }
+
   public postConfig: RequestHandler = async (req, res) => {
     const currentEnabledPrisons = await this.prisonerService.getPrisonsWithServiceCode(req.body.apiId)
     const checkedBoxes = [req.body.checkedBoxes === undefined ? [] : req.body.checkedBoxes].flat()
     const redirectSection = readOnlyNomisScreens.find(it => it.apiId === req.body.apiId).id
 
-    this.enableCheckedBoxes(checkedBoxes, currentEnabledPrisons, req.body.apiId)
-    this.disableUncheckedBoxes(checkedBoxes, currentEnabledPrisons, req.body.apiId)
+    const newReadOnly = this.enableCheckedBoxes(checkedBoxes, currentEnabledPrisons, req.body.apiId)
+    const newUnchecked = this.disableUncheckedBoxes(checkedBoxes, currentEnabledPrisons, req.body.apiId)
 
-    res.redirect(`/config#${redirectSection}`)
+    res.redirect(
+      `/config/update?id=${redirectSection}&readonly=${newReadOnly.join(',')}&notreadonly=${newUnchecked.join(',')}`,
```

Comment:
couldn't you just go directly to the `/config?id=${redirectSection}&readonly=${newReadOnly.join(',')}&notreadonly=${newUnchecked.join(',')}` route here and have query parameters logic? Saves a redirect and using flash?

Reply:
Thanks, I've updated to utilise this